sklearnex/basic_statistics/incremental_basic_statistics.py

```diff
@@ -152,9 +153,11 @@ def _onedal_partial_fit(self, X, weights, queue):
             X = check_array(
                 X,
                 dtype=[np.float64, np.float32],
-                copy=self.copy_X,
             )
 
+        if sample_weight is not None:
+            sample_weight = _check_sample_weight(sample_weight, X)
```

Comment:
If I am not wrong we are already doing such sample weight checks on onedal side

Reply:
Unfortunately it was missed when IncrementalBasicStatistics was pulled in recently. This check guarantees that the supported datatype is some flavor of numpy for dtype checks, which was uncovered in the Pandas dataframe support PR. This is also causing issues with fp32 support for GPU,